File: practices/zuma/ez.py
zuma = []
zn = []
nn = []

nm = input()
n, m = nm.split()
n = int(n)
m = int(m)

for i in range(n):
    ball = int(input())
    zuma.append(ball)

for x in range(len(zuma)):
    zn.append(x+1)

ll = len(zn)

for j in range(m):
    cl = input()
    c, l = cl.split()
    nn.append(ll+j+1)
    zn.insert((zn.index(int(l)))+1, nn[j])

for i in zn:
    print(i)

File: practices/zuma/ez.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int main() {
    vector<int> zuma;
    vector<int> zn;
    vector<int> nn;

    int n, m;
    cin >> n >> m;

    for (int i = 0; i < n; i++) {
        int ball;
        cin >> ball;
        zuma.push_back(ball);
    }

    for (int i = 0; i < n; i++) {
        zn.push_back(i + 1);
    }

    int ll = zn.size();

    for (int j = 0; j < m; j++) {
        int c, l;
        cin >> c >> l;

        nn.push_back(ll + j + 1);

        auto it = find(zn.begin(), zn.end(), l);
        if (it != zn.end()) {
            zn.insert(it + 1, nn[j]);
        }
    }

    for (int i = 0; i < zn.size(); i++) {
        cout << zn[i] << endl;
    }

    return 0;
}
